### PR TITLE
Feature/condo/SBERDOMA-108/add_behavior_recorder

### PR DESCRIPTION
**Pull request by leonid-d**
https://www.plerdy.com/

### CODE REVIEW

File: apps/condo/domains/common/components/containers/BehaviorRecorder.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+const BehaviorRecorder = () => {
+    return (
+        <div dangerouslySetInnerHTML={{
+            __html: `<script type="text/javascript" defer>
+                var _protocol = (("https:" == document.location.protocol) ? " https://" : " http://");
+                var _site_hash_code = "4b23250e438223063d2fb9af042a8199";
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/BehaviorRecorder.tsx:8]: from env in JSON format like FIREBASE_CONFIG

Comment (Dimitreee) [reply apps/condo/domains/common/components/containers/BehaviorRecorder.tsx:8]: +

Comment (Dimitreee) [reply apps/condo/domains/common/components/containers/BehaviorRecorder.tsx:8]: ++

**Review by Dimitreee — APPROVED** [review]

**Review by pahaz — CHANGES_REQUESTED** [review]
NO it's bad practice to add HTML Injection from .env %)

**Comment by pahaz** [comment]
You can look at `ADDRESS_SUGGESTIONS_CONFIG` as an example

**Comment by AntonAL** [comment]
I meant to implement it like this:

.env.example:

```
BEHAVIOR_RECORDER_PLERDY='{site_hash_code: 4b23250e438223063d2fb9af042a8199, suid: 17358}'
```

apps/condo/domains/common/components/containers/BehaviorRecorder.tsx 

```jsx
type BehaviourRecordingEngine = 'plerdy' | 'yandex'

type Props = {
	engine: BehaviourRecordingEngine;
}

const BehaviorRecorder = ({ engine }: Props) => {
     const { publicRuntimeConfig } = getConfig()
     const { behaviorRecorderParams } = publicRuntimeConfig // expected `{ plerdy: …, yandex: … }`

     const params = parseParamsFor[engine](behaviorRecorderParams[engine])

     return (
         <div dangerouslySetInnerHTML={{
             __html: injectParamsFor[engine](params),
         }}>
         </div>
     )
}

type PlerdyParams = {
	site_hash_code: String;
	suid: String;
}

// Types for params of other engines
// ...

const parseParamsFor = {
	plerdy: (params: String): PlerdyParams => {
		const SITE_HASH_CODE_REGEXP = /\b([a-z0-9]{32})\b/
		const SUID_REGEXP = /\b([0-9]{5})\b/

		const plerdyParams = JSON.parse(params)
		
		let site_hash_code
		
		if (plerdyParams.site_hash_code.match(SITE_HASH_CODE_REGEXP)) {
			site_hash_code = plerdyParams.site_hash_code
		} else {
			throw new Error('Incorrect value of site_hash_code param')
		}

		let suid

		if (plerdyParams.suid.match(SUID_REGEXP)) {
			suid = plerdyParams.suid
		} else {
			throw new Error('Incorrect value of suid param')
		}

		return { site_hash_code, suid }
	},
	// yandex: () => 
	/ ...
}


const injectParamsFor = {
	plerdy: ({ site_hash_code, suid }: PlerdyParams) => (
		`<script type="text/javascript" defer>var _protocol = (("https:" == document.location.protocol) ? " https://" : " http://");var _site_hash_code = "${site_hash_code}";var _suid = ${suid};</script><script type="text/javascript" defer src="https://a.plerdy.com/public/js/click/main.js"></script>`
	),
	// yandex : () => 
	// ...
}
```


Mount like this:

```jsx
<BehaviorRecorder engine="plerdy"/>
```

Runtime errors will be catched by `ErrorBoundary`.

**Review by dkoviazin — APPROVED** [review]
OK

**Pull request closed without merging**